wombat/src/main/include/drivetrain/SwerveDrive.h

```diff
@@ -1,38 +1,124 @@
-/* #pragma once */
-/**/
-/* #include "behaviour/Behaviour.h" */
-/* #include "subsystems/Subsystem.h" */
-/* #include "utils/PID.h" */
-/**/
-/* namespace wom { */
-/*     enum ModuleName { */
-/*         FrontLeft, */
-/*         FrontRight, */
-/*         BackLeft, */
-/*         BackRight, */
-/*     }; */
-/**/
-/*     struct SwerveModuleConfig {}; */
-/**/
-/*     enum SwerveModuleState { */
-/*         kIdle, */
-/*         kCalibration, */
-/*         kPID, */
-/*     }; */
-/**/
-/*     class SwerveModule : public wom::Subsystem<SwerveModuleConfig, SwerveModuleState>, public behaviour::HasBehaviour { */
-/*       public: */
-/*           SwerveModule(wom::Subsystem<SwerveModuleConfig, SwerveModuleState> _, wom::ModuleName name, wom::SwerveModuleConfig config, wom::SwerveModuleState state); */
-/*           ~SwerveModule(); */
-/**/
-/*           void OnStart() override; */
-/*           void OnUpdate(units::second_t dt) override; */
-/**/
-/*       protected: */
-/**/
-/*       private: */
-/*           wom::ModuleName _name; */
-/**/
-/*     }; */
-/**/
-/* }; */
+#pragma once
+
+#include <networktables/NetworkTableInstance.h>
+#include <networktables/NetworkTable.h>
+#include <networktables/DoubleTopic.h>
+
+#include <units/angular_velocity.h>
+#include <units/charge.h>
+#include <units/math.h>
+#include <units/time.h>
+#include <units/velocity.h>
+
+#include <iostream>
+#include <string>
+#include <memory>
+
+#include "behaviour/HasBehaviour.h"
+#include "utils/Gearbox.h"
+#include "utils/PID.h"
+#include "utils/Util.h"
+#include "vision/Limelight.h"
+
+namespace wom {
+namespace drivetrain {
+  enum class SwerveModuleName {
+    FrontLeft,
+    FrontRight,
+    BackLeft,
+    BackRight,
+  };
+
+  struct SwerveModuleConfig {
+    wom::utils::Gearbox rotationGearbox;
+    wom::utils::Gearbox movementGearbox;
+
+    wom::utils::PIDConfig<units::radians_per_second, units::volt> rotationalVelocityPID;
+    wom::utils::PIDConfig<units::meters_per_second, units::volt> movementVelocityPID;
+
+    wom::utils::PIDConfig<units::radian, units::radians_per_second> rotationPID;
+    wom::utils::PIDConfig<units::meter, units::meters_per_second> movementPID;
+
+    units::meter_t wheelRadius;
+
+    std::string path;
+    SwerveModuleName name;
+  };
+
+  enum class SwerveModuleState {
```

Comment:
you could also add a manualTurn / manualDrive mode here that might be useful for testing

Reply:
Yeah sounds good fr fr